extensions/wikia/CuratedContent/CuratedContentHelper.class.php

```diff
@@ -0,0 +1,163 @@
+
+		foreach ( $section['items'] as &$item ) {
+			$this->fillItemInfo( $item );
+		}
+
+		return $section;
+	}
+
+	private function fillItemInfo( &$item ) {
+		$title = Title::newFromText( $item['title'] );
+		if ( $title instanceof Title && $title->exists() ) {
+			$articleId = $title->getArticleId();
+			$namespaceId = $title->getNamespace();
+			$imageId = (int)$item['image_id'];
+
+			$item['type'] = self::getTypeFromNamespaceId( $namespaceId );
+
+			switch ( $item['type'] ) {
+				case self::STR_FILE:
+					$info = self::getVideoInfo( $title );
+					if ( !empty( $info ) ) {
+						$item['type'] = self::STR_VIDEO;
+						$item['video_info'] = $info;
+					}
+					break;
+
+				case self::STR_CATEGORY:
+					$category = Category::newFromTitle( $title );
+					if ( $category instanceof Category && $category->getID() ) {
+						$count = $category->getPageCount();
+						if ( empty( $count ) ) {
+							$item['type'] = self::STR_EMPTY_CATEGORY;
+						}
+					} else {
+						$item['type'] = null;
+					}
+					break;
+			}
+
+			if ( empty( $imageId ) ) {
+				$imageTitle = self::findFirstImageTitleFromArticle( $articleId );
+				if ( !empty( $imageTitle ) ) {
+					$imageId = $imageTitle->getArticleId();
+				}
+			}
+			$item['article_id'] = $articleId;
+			$item['image_id'] = $imageId;
+		}
+	}
+
+	public static function getIdFromCategoryName( $categoryName ) {
+		$category = Title::newFromText( $categoryName, NS_CATEGORY );
+
+		return ($category instanceof Title && $category->exists()) ? $category->getArticleID() : 0;
+	}
+
+	public static function getImageUrl( $id, $imageSize = 50 ) {
+		$thumbnail = (new ImageServing( [ $id ], $imageSize, $imageSize ))->getImages( 1 );
+
+		return !empty( $thumbnail ) ? $thumbnail[$id][0]['url'] : '';
+	}
+
+	private function getVideoInfo( $title ) {
+		$mediaService = new MediaQueryService();
+		$mediaInfo = $mediaService->getMediaData( $title );
+		if ( !empty( $mediaInfo ) ) {
+			if ( $mediaInfo['type'] === MediaQueryService::MEDIA_TYPE_VIDEO ) {
+				return [
+					'provider' => $mediaInfo['meta']['provider'],
+					'thumb_url' => $mediaInfo['thumbUrl'],
+					'videoId' => $mediaInfo['meta']['videoId']
+				];
+			}
+		}
+		return null;
+	}
+
+	public static function findImage( $imageId, $articleId = 0 ) {
+		$url = '';
+		$imageTitle = null;
+
+		if ( empty( $imageId ) ) {
+			$imageId = null;
+			$imageTitle = self::findFirstImageTitleFromArticle( $articleId );
+		} else {
+			$imageTitle = Title::newFromID( $imageId );
+		}
+		if ( $imageTitle instanceof Title && $imageTitle->exists() ) {
+			$url = self::getUrlFromImageTitle( $imageTitle );
+			$imageId = $imageTitle->getArticleId();
+		}
+
+		return [ $imageId, $url ];
+	}
+
+	public static function findFirstImageTitleFromArticle( $articleId ) {
+		$imageTitle = null;
+		if ( !empty( $articleId ) ) {
+			$imageServing = new ImageServing( [ $articleId ] );
+			$image = $imageServing->getImages( 1 );
```

Comment:
Is this method filtering out small images e.g. currency  and skill icons or icons in notices e.g. stub info and spoiler alert?